operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Note: make sure the input image has the format BGR
+
+#include "draw_bounding_box.hpp"
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cassert>
+#include <gsl/narrow>
+#include <gsl/span>
+#include <gsl/span_ext>
+#include <unordered_map>
+#include <vector>
+#include "exceptions.h"
+
+namespace ort_extensions {
+namespace {
+constexpr std::array<std::array<uint8_t, 3>, 10> KBGRColorMap = {{
+    {{0, 0, 255}},      // Red
+    {{0, 255, 0}},      // Green
+    {{255, 0, 0}},      // Blue
+    {{255, 255, 0}},    // Cyan
+    {{255, 0, 255}},    // Magenta
+    {{0, 0, 128}},      // Dark Red/Maroon
+    {{0, 128, 0}},      // Dark Green/Lime
+    {{128, 0, 0}},      // Dark Blue/Navy
+    {{0, 0, 0}},        // Black
+    {{128, 128, 128}},  // Gray
+}};
+
+// Used to store image data, width, height, and number of channels
+struct ImageView {
+  gsl::span<uint8_t> data;
+  int64_t height;
+  int64_t width;
+  int64_t channels;
+};
+
+static constexpr size_t kBoxClassIndex = 5;
+static constexpr size_t kBoxScoreIndex = 4;
+
+// To represent Boxes, [num of boxes, box_info]
+// box_info = [x1, y1, x2, y2, score, class]
+class BoxArray {
+ private:
+  void SortBoxesByScore(gsl::span<const float> data) {
+    boxes_by_score_.reserve(NumBoxes());
+    for (size_t i = 0; i < NumBoxes(); ++i) {
+      boxes_by_score_.push_back(data.subspan(i * shape_[1], shape_[1]));
+    }
+
+    std::sort(boxes_by_score_.begin(), boxes_by_score_.end(),
+              [](const gsl::span<const float>& first, const gsl::span<const float>& second) {
+                return first[kBoxScoreIndex] > second[kBoxScoreIndex];
+              });
+  }
+
+ public:
+  BoxArray(const std::vector<int64_t>& shape, gsl::span<const float> data, BoundingBoxFormat bbox_mode)
+      : shape_(shape), bbox_mode_(bbox_mode) {
+    SortBoxesByScore(data);
+  }
+
+  [[nodiscard]] gsl::span<const float> GetBox(size_t index) const {
+    assert(index < boxes_by_score_.size());
+    return boxes_by_score_[index];
+  }
+
+  [[nodiscard]] int64_t NumBoxes() const {
+    return shape_[0];
+  }
+
+  auto BBoxMode() const {
```

Comment:
nit: using 'auto' in the public interface of a class is somewhat painful for the reader. they have to go find the definition of the member variable being returned in order to figure out the type. i.e. you're unnecessarily forcing them to work harder to understand the interface of your class.